### ISSUE TITLE
Add `packageOptions` config value for packages mode

### ISSUE THREAD

**Issue by Gerrit0** [TypeStrong/typedoc#2523]
## Search Terms

Configuration duplication, packages mode

## Problem

Currently, to set up TypeDoc with "packages" mode, you have to set up a config file for each package. In my experience, these generally look something like the following... in every single package folder.

```json
{
    "extends": ["../../typedoc.base.json"],
    "entryPoints": ["src/index.ts"]
}
```

## Suggested Solution

Support a `packageOptions` section when reading an options file. This value should be an object which recursively references TypeDoc's base configuration schema so that monorepos containing many packages whose TypeDoc configuration is currently forced to be duplicated in each directory can instead be present only in the root config, then applied to each package's options object before reading any options for that directory.

**Comment by nicfv** [comment]
I notice that tsconfig also needs to be present in every package for the `rootdir`/`outdir`/`include` paths. Do you know if there's currently a way around this?

**Comment by Gerrit0** [comment]
That's a limitation of TypeScript so far as I know...

**Comment by Gerrit0** [comment]
Included in TypeDoc 0.26, which is releasing 2024/06/21
